src/test/java/com/shapesecurity/salvation/PolicyQueryingTest.java

```diff
@@ -69,11 +69,32 @@ public void testAllowsFromSource() throws ParseException, TokeniserException {
         assertFalse("resource is not allowed", p.allowsImgFromSource(URI.parse("https://abc.am")));
         assertFalse("resource is not allowed", p.allowsStyleFromSource(URI.parse("ftp://www.abc.am:555")));
         assertFalse("resource is not allowed", p.allowsScriptFromSource(URI.parse("https://www.def.am:555")));
+        assertFalse("resource is not allowed", p.allowsFrameFromSource(URI.parse("https://www.def.am:555")));
+        assertFalse("resource is not allowed", p.allowsChildFromSource(
+            URI.parse("https://www.def.am:555")));
+
 
         p = Parser.parse("default-src *:*", "http://abc.com");
         assertTrue("resource is allowed", p.allowsImgFromSource(URI.parse("http://abc.am")));
         assertFalse("resource is not allowed", p.allowsScriptFromSource(URI.parse("https://www.def.am:555")));
         assertFalse("resource is not allowed", p.allowsStyleFromSource(URI.parse("ftp://www.abc.am:555")));
+
+        p = Parser.parse("frame-src http:;", URI.parse("https://abc.com"));
```

Comment:
Use a `default-src 'none'` here and then assert that child sources aren't allowed.